scarlet/lite/initialization.py

```diff
@@ -318,103 +318,159 @@ def init_fista_component(center, bbox, sed, morph, observation, bg_thresh=None):
+        self.psf_sed = self.convolved_psf[:, self.py, self.px]
+        # Set the input parameters
+        self.min_snr = min_snr
+        self.use_mask = use_mask
+        self.percentile = percentile
+        self.thresh = thresh
+
+
+def init_main_source(center, init):
+    """
+    """
+    # Calculate the signal to noise at the center of this source
+    snr = np.floor(calculate_snr(
+        init.observation.images,
+        init.observation.variance,
+        init.observation.psfs,
+        center,
+    ))
+    component_snr = snr / init.min_snr
+
+    # Initialize the bbox, morph, sed for a single component source
+    bbox, morph, sed = bbox, morph, sed = init_main_parameters(
+        init.detect, center, init.observation, init.convolved, init.use_mask, init.thresh)
+
+    if morph is None:
+        # There wasn't sufficient flux for an extended source,
+        # so create a PSF source.
+        sed_center = (slice(None), center[0], center[1])
+        sed = init.observation.images[sed_center] / init.psf_sed
+        sed[sed < 0] = 0
+        morph = init.model_psf.copy()
+        morph = morph/np.max(morph)
+        bbox = Box(init.model_psf.shape, origin=(center[0]-init.py, center[1]-init.px))
+        components = [LiteComponent(center, init.observation.bbox[0] @ bbox, sed, morph)]
+    elif component_snr >= 2:
+        # There was enough flux for a 2-component source,
+        # so split the single component model into two components, using the
+        # same algorithm as scarlet main.
+        bulge_morph = morph.copy()
+        disk_morph = morph
+        flux_thresh = init.percentile / 100
+        mask = disk_morph > flux_thresh
+        disk_morph[mask] = flux_thresh
+        bulge_morph -= flux_thresh
+        bulge_morph[bulge_morph < 0] = 0
+
+        if bulge_morph is None or disk_morph is None:
+            if bulge_morph is None:
+                if disk_morph is None:
+                    return None
+                morph = disk_morph
+            else:
+                morph = bulge_morph
+            # One of the components was null,
+            # so initialize as a single component
+            components = [LiteComponent(center, init.observation.bbox @ bbox, sed, morph)]
+        else:
+            bulge_morph /= np.max(bulge_morph)
+            disk_morph /= np.max(disk_morph)
+
+            bulge_sed, disk_sed = multifit_seds(init.observation, [bulge_morph, disk_morph], [bbox, bbox])
+
+            components = [
+                LiteComponent(center, init.observation.bbox[0] @ bbox, bulge_sed, bulge_morph),
+                LiteComponent(center, init.observation.bbox[0] @ bbox, disk_sed, disk_morph),
+            ]
+    else:
+        components = [LiteComponent(center, init.observation.bbox[0] @ bbox, sed, morph)]
+
+    return LiteSource(components, init.observation.dtype)
+
+
 def init_all_sources_main(observation, centers, detect=None,
         min_snr=50, use_mask=False, percentile=25, thresh=0.5):
```

Comment:
You seem to be ignoring most of these parameters instead of passing them to `Chi2InitParameters`.

Reply:
Great catch, that would have been unfortunate to include.